earthfile2llb/converter.go

```diff
@@ -1594,6 +1595,71 @@ func (c *Converter) Pipeline(ctx context.Context) error {
 	return nil
 }
 
+func (c *Converter) ExpandWildcard(ctx context.Context, fullTargetName string, cmd spec.Command) ([]spec.Command, error) {
+
+	parsedTarget, err := domain.ParseTarget(fullTargetName)
+	if err != nil {
+		return nil, err
+	}
+
+	if strings.Contains(fullTargetName, "**") {
+		return nil, errors.New("globstar (**) pattern not yet supported")
+	}
+
+	var target domain.Target
+	if c.target.IsRemote() {
+		target = c.target
+	} else {
+		target = parsedTarget
+	}
+
+	matches, err := c.opt.Resolver.ExpandWildcard(ctx, c.opt.GwClient, c.platr, target, parsedTarget.GetLocalPath())
+	if err != nil {
+		return nil, err
+	}
+
+	children := []spec.Command{}
+	for _, match := range matches {
+		childTargetName := fmt.Sprintf("./%s+%s", match, parsedTarget.GetName())
+
+		childTarget, err := domain.ParseTarget(childTargetName)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to parse target %q", childTargetName)
+		}
+
+		data, _, _, err := c.ResolveReference(ctx, childTarget)
+		if err != nil {
+			notExist := buildcontext.ErrEarthfileNotExist{}
```

Comment:
Just realizing another edge case. What if the Earthfile does exist, but it does not contain the target we're looking for? Should we also skip that error?

Reply:
IIRC `ExpandWildcard` & `ResolveReference` do not verify that the target exists, but there's a check on L1639 which does. There are also tests for this scenario.

Reply:
Oh yeah - makes sense!